### ISSUE TITLE
Consider browser-based / Cucumber testing

### ISSUE THREAD

**Issue by beechnut** [18F/NRM-Grants-Agreements#335]
We should consider whether to set up comprehensive BDD/Cucumber-style tests to support technology stack transition, should a decision be made to continue with a Python prototype but eventually have a Java-based stack.

One piece of value we plan on delivering is a comprehensive test suite. We are already expecting to provide unit tests and integration tests. Adding Cucumber to the mix allows us to provide language-agnostic tests which can be re-implemented in the language of the final application stack.

Cucumber tests are written in plain English ([source](https://cucumber.io/docs/gherkin/reference/)) like the following, before being implemented as code:

```
Feature: Guess the word

  Scenario: Maker starts a game
    When the Maker starts a game
    Then the Maker waits for a Breaker to join
```

**We'll know we're done when...**
- [ ] A decision has been made re: whether to implement Cucumber tests.

**Tasks**
- [ ] Determine the technology stack for the prototyping and vendor transition phases.
- [ ] Come to consensus on the value of being able to pass on Cucumber-style tests to a vendor.

**Comment by aaronburk** [comment]
Thanks for creating this Matt ( I have forwarded it to Adam Shepherd as well) not sure if he onboarded and getting these notifications yet

**Comment by beechnut** [comment]
In doing some [initial research](https://stackoverflow.com/questions/9485962/does-python-have-anything-like-capybara-cucumber) I found the following:

- [A 2011 article on integration testing in Django](https://www.cilliano.com/2011/02/07/django-bdd-with-lettuce-and-splinter.html) which led to a number of useful libraries
- [Splinter](https://github.com/cobrateam/splinter), for simulating a browser
- [Salad](https://github.com/salad/salad), a package that combines multiple useful packages for integration testing
- [behave-django](https://github.com/behave/behave-django), which is another library supporting integration/BDD testing.
- [Capybara, ported to Python!](https://elliterate.github.io/capybara.py/)
- [Helium](https://github.com/mherrmann/selenium-python-helium), a wrapper for Selenium
- [SeleniumLibrary](https://github.com/robotframework/SeleniumLibrary), another wrapper for Selenium

**Comment by beechnut** [comment]
[See this comment for some research on integration testing tools](https://github.com/USDAForestService/NRM-Grants-Agreements/issues/289#issuecomment-875046829), which includes some BDD/Cucumber/Gherkin-type tools.
